pyat/at/tracking/particles.py

```diff
@@ -4,92 +4,116 @@
 import numpy
 from at.physics import ohmi_envelope
 from at.lattice.lattice_object import Lattice
+from scipy.constants import c as clight
 
 __all__ = ['beam', 'sigma_matrix']
 
+def _generate__sigma_matrix(ld0, emitx, emity, blength, espread, radiation):
+    if radiation:
+        sig_matrix = emitx*ld0.R[0] + emity*ld0.R[1] + blength*espread*ld0.R[2]
 
-def sigma_matrix(argin):
+    else:
+        sig_matrix_long = numpy.array([
+                             [espread*espread, 0],
+                             [0, blength*blength]
+                                      ])
+
+        sig_matrix_trans = emitx*ld0.R[0] + emity*ld0.R[1] 
+
+        sig_matrix = numpy.block([
+                        [sig_matrix_trans[:4,:4],
+                            numpy.zeros((4, 2))],
+                        [numpy.zeros((2, 4)),
+                            sig_matrix_long]
+                                ])
+    return sig_matrix
+
+def _compute_bunch_length_from_espread(ring, espread):
+    emass = 0.510998910e6
+    gamma = ring.energy/emass
+    beta = numpy.sqrt(1.0-1.0/gamma/gamma)
+    radFlag = ring.radiation
+        
+    ring.radiation_off()
+    mcf = ring.get_mcf()
+    ring.radiation_on()
+    envel = ring.envelope_parameters()
+
+    if not radFlag:
+        ring.radiation_off()
```

Comment:
One cannot expect that calling `sigma_matrix(ring,..` modifies the ring. So it should be:
```python
mcf = ring.radiation_off(copy=True).get_mcf()
envelope = ring.radiation_on(copy=True).envelope_parameters()
```
or 
```python
mcf = get_mcf(ring.radiation_off(copy=True)
envelope = envelope_parameters(ring.radiation_on(copy=True))
```
as you prefer.

In addition, turning radiation off and then on does not ensure that you come back to the initial state (which multipoles did the user activate?). So it should be avoided.